tests/e2e.test.ts

```diff
@@ -384,5 +385,45 @@ describe('End-to-end tests', () => {
         `)
       })
     })
+    describe('--error-format option', () => {
+      it('fails with an invalid error format', async () => {
+        const invalidErrorFormat: string = 'invalid-format'
+
+        const checkOutput = await cli(
+          `check --error-format ${invalidErrorFormat}`,
+          basicTsErrorOutput
+        )
+        expect(checkOutput.code).toBe(1)
+        expect(checkOutput.stderr).toMatch(
+          `error: option '--error-format [error-format]' argument '${invalidErrorFormat}' is invalid.`
+        )
+      })
+
+      it('has expected gitlab output', async () => {
```

Comment:
You are missing tests for "no new errors".
I am not sure what the desired output for gitlab would be then. An empty json? 

@TimMikeladze Are tests for `--ignore-message` changes the way error summaries are collected, so there should probably be tests for that combination too, right?

Reply:
@func0der GitLab expects an empty array. I have added another test for "no new errors".